amqprs/src/api/channel/mod.rs

```diff
@@ -124,6 +124,8 @@ pub struct Channel {
 pub(crate) struct SharedChannelInner {
     /// open state
     is_open: AtomicBool,
+    /// prefetch count
+    prefetch_count: AtomicU16,
```

Comment:
as above comments/reason, I would suggest to remove all pretech_count logic in this PR. We can have another PR for more discussion if there is a proper solution for your intention.

Reply:
I think this topic is linked to what I'm currently doing, imo it makes to do it in this PR. You tell me once you've read my above reply :)

Reply:
I do believe the library level buffer size is linked to server side settings that you define with `basic_qos`, but I may be mistaken.

Reply:
> I do believe the library level buffer size is linked to server side settings that you define with `basic_qos`, but I may be mistaken.

It is related in some level of course,  but no guarantee in all use cases due to concurrent nature of the library.